src/drive/web/modules/views/Recent/index.spec.jsx

```diff
@@ -0,0 +1,126 @@
+import { Router, hashHistory, Route } from 'react-router'
+
+import { setupStoreAndClient } from 'test/setup'
+import { generateFile } from 'test/generate'
+import AppLike from 'test/components/AppLike'
+
+import RecentViewWithProvider from './index'
+import { useFilesQueryWithPath } from './useFilesQueryWithPath'
+
+jest.mock('cozy-client/dist/hooks/useQuery', () => jest.fn())
+jest.mock('./useFilesQueryWithPath', () => ({
+  ...jest.requireActual('./useFilesQueryWithPath'),
+  useFilesQueryWithPath: jest.fn()
+}))
+
+describe('Recent View', () => {
+  const setup = ({
+    nbFiles = 5,
+    totalCount,
+    useQueryResultAttributes,
+    path = '/test',
+    dir_id = 'io.cozy.files.root-dir',
+    updated_at = '2020-05-14T10:33:31.365224+02:00'
+  } = {}) => {
+    const { store, client } = setupStoreAndClient({
+      selection: {
+        selected: {}
+      }
+    })
+
+    client.plugins.realtime = {
+      subscribe: jest.fn(),
+      unsubscribe: jest.fn()
+    }
+    client.query = jest.fn().mockReturnValue([])
+    client.stackClient.fetchJSON = jest
+      .fn()
+      .mockReturnValue({ data: [], rows: [] })
+    const filesFixture = Array(nbFiles)
+      .fill(null)
+      .map((x, i) =>
+        generateFile({ i, type: 'file', path, dir_id, updated_at })
+      )
+
+    useQuery.mockReturnValue({
+      data: filesFixture,
+      count: totalCount || filesFixture.length,
+      fetchMore: jest.fn().mockImplementation(() => {
+        throw new Error('Fetch more should not be called')
+      }),
+      ...useQueryResultAttributes
+    })
+    const filesFixtureWithPath = {
+      data: filesFixture.map(f => {
+        return {
+          ...f,
+          displayedPath: path
+        }
+      })
+    }
+    useFilesQueryWithPath.mockReturnValue(filesFixtureWithPath)
+    const rendered = render(
+      <AppLike client={client} store={store}>
+        <Router history={hashHistory}>
+          <Route path="/" component={RecentViewWithProvider} />
+          <Route path="/folder/:id" component={() => null} />
+        </Router>
+      </AppLike>
+    )
+    return { ...rendered, client }
+  }
+
+  it('test the recent view', async () => {
+    const nbFiles = 2
+    const path = '/test'
+    const dir_id = 'dirIdParent'
+    const updated_at = '2020-05-14T10:33:31.365224+02:00'
+    const { getByText } = setup({
+      nbFiles,
+      path,
+      dir_id,
+      updated_at
+    })
+    const getByTextWithMarkup = text => {
+      getByText((content, node) => {
+        const hasText = node => node.textContent === text
+        const childrenDontHaveText = Array.from(node.children).every(
+          child => !hasText(child)
+        )
+        return hasText(node) && childrenDontHaveText
+      })
+    }
+    //Get the HTMLElement containing the filename
+    const el0 = getByText(`foobar0`)
+    //check if the filename is displayed with the extension
+    expect(getByTextWithMarkup(`foobar0.pdf`))
```

Comment:
It seems an expectation is missing here.

```
expect(..).toBeDefined() # ?
```

Reply:
`getBy*` throw when the predicate was not found.  So if it's doesn't throw, it works. Let's remove the expect() here it doesn't do anything, you right.

Reply:
You can write `expect(() => getByTextWithMarkup('foobar0.pdf')).not.toThrow()`.